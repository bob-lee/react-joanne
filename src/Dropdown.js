import React from 'react';
import { NavLink } from 'react-router-dom'

export default class Dropdown extends React.Component {
  state = { checked: false }

  // constructor(props) {
  //   super(props)
  // }

  handleInputChange = event => {
    //console.log('handleInputChange', event.target)
    this.setState({ checked: true })
  }

  handleOverlay = () => {
    console.log('overlay')
    this.setState({ checked: false })
  }

  render() {
    const classes = `ddm ${this.props.title === 'work' ? '' : 'active'}`
    return (
      <div className={classes} to="/work">
        <input type="checkbox" id="ddm-1"
          checked={this.state.checked}
          onChange={this.handleInputChange} />
        <label htmlFor="ddm-1" className="header-menu-item">{this.props.title}</label>
        <i className="fa fa-angle-right fa-lg" aria-hidden="true"></i>
        <div className="dd" onClick={this.handleOverlay}>
          <NavLink activeClassName="active" to="/work/portrait">portrait</NavLink>
          <NavLink activeClassName="active" to="/work/painting">painting</NavLink>
          <NavLink activeClassName="active" to="/work/illustration">illustration</NavLink>
        </div>
        <div className="overlay" onClick={this.handleOverlay}>
        </div>
      </div>
    )
  }
}