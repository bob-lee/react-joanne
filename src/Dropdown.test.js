import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';

const renderWithTitle = (title) => {
  return renderer
    .create(<Router><Dropdown title={title} /></Router>)
    .toJSON();
}

it('Dropdown renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Router><Dropdown title="work" /></Router>, div);
});
/*
it('Dropdown renders correctly for non-work page', () => {
  expect(renderWithTitle('work')).toMatchSnapshot();
});

it(`Dropdown renders correctly for '/work/painting' page`, () => {
  expect(renderWithTitle('painting')).toMatchSnapshot();
});
*/
it(`shouldn't have 'active' class for non-work page`, () => {
  const wrapper = shallow(<Dropdown title="work" />);
  //console.log(wrapper.props());

  expect(wrapper.props().className).not.toMatch(/active/)
  expect(toJson(wrapper)).toMatchSnapshot();
});

it(`should have 'active' class for '/work/painting' page`, () => {
  const wrapper = shallow(<Dropdown title="painting" />);
  //console.log(wrapper.props());

  expect(wrapper.props().className).toMatch(/active/)
  expect(toJson(wrapper)).toMatchSnapshot();
});

it(`initially should be unchecked`, () => {
  const wrapper = shallow(<Dropdown title="work" />);
  expect(wrapper.state().checked).toBe(false);
});

it(`should be checked on label-click`, () => {
  const wrapper = shallow(<Dropdown title="work" />);
  wrapper.find('input').simulate('change');
  expect(wrapper.state().checked).toBe(true);
});

it(`should be unchecked on submenu-click`, () => {
  const wrapper = shallow(<Dropdown title="work" />);
  //wrapper.find('input').simulate('change');
  wrapper.find('.dd').simulate('click');
  expect(wrapper.state().checked).toBe(false);
});

it(`should be unchecked on overlay-click`, () => {
  const wrapper = shallow(<Dropdown title="work" />);
  //wrapper.find('input').simulate('change');
  wrapper.find('.overlay').simulate('click');
  expect(wrapper.state().checked).toBe(false);
});
