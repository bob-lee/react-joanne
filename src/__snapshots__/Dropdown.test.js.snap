// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have 'active' class for '/work/painting' page 1`] = `
<div
  className="ddm active"
  to="/work"
>
  <input
    checked={false}
    id="ddm-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="header-menu-item"
    htmlFor="ddm-1"
  >
    painting
  </label>
  <i
    aria-hidden="true"
    className="fa fa-angle-right fa-lg"
  />
  <div
    className="dd"
    onClick={[Function]}
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/portrait"
    >
      portrait
    </NavLink>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/painting"
    >
      painting
    </NavLink>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/illustration"
    >
      illustration
    </NavLink>
  </div>
  <div
    className="overlay"
    onClick={[Function]}
  />
</div>
`;

exports[`shouldn't have 'active' class for non-work page 1`] = `
<div
  className="ddm "
  to="/work"
>
  <input
    checked={false}
    id="ddm-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="header-menu-item"
    htmlFor="ddm-1"
  >
    work
  </label>
  <i
    aria-hidden="true"
    className="fa fa-angle-right fa-lg"
  />
  <div
    className="dd"
    onClick={[Function]}
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/portrait"
    >
      portrait
    </NavLink>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/painting"
    >
      painting
    </NavLink>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/work/illustration"
    >
      illustration
    </NavLink>
  </div>
  <div
    className="overlay"
    onClick={[Function]}
  />
</div>
`;
